tools/autograd/gen_variable_type.py

```diff
@@ -117,6 +117,25 @@
 CALL_VIA_DERIVED = CodeTemplate("""\
 baseType->${method_prefix_derived}${base_name}(${unpacked_args})""")
 
+# If the `baseType` operation has return values, we use the `tmp` variable to hold the
+# values temporarily and pass the values to the return variables outside of the
+# `at::AutoNonVariableTypeMode` guard block.
+DISPATCH_TO_NON_VAR_TYPE_WITH_RETURN_VALUES = CodeTemplate("""\
+decltype(${base_type_call}) tmp;
```

Comment:
The downside of doing it this way is this code does not work if the type of `base_type_call` does not have a default constructor. With lambdas, you can write this without requiring a default constructor:

```
auto result = ([&]() {
  at::AutoNonVariableTypeMode non_var_type_mode(true);
  return ${base_type_call};
})();
```